### PR TITLE
Ticket 38: modal and delete product

### PR DESCRIPTION
**Pull request by maximFeldman1**
build modal and delete product
Screen Shot 2022-12-27 at 17 39 09
Screen Shot 2022-12-27 at 17 38 54

### CODE REVIEW

File: src/Pages/Products/components/ProductsTable/index.tsx [hunk 1]
```diff
@@ -3,16 +3,24 @@ import { AgGridReact } from "@ag-grid-community/react";
...
 `;
 
 export const ProductsTable = () => {
-  const { data: products } = useQuery("getAllProducts", ProductService.getAll);
-
+  const { data: products, refetch } = useQuery(
+    "getAllProducts",
+    ProductService.getAll
+  );
+  // refetch("getAllProducts");
```
Comment (amitmarko) [inline src/Pages/Products/components/ProductsTable/index.tsx:23]: remove

Comment (amitmarko) [inline src/Pages/Products/components/ProductsTable/index.tsx]: remove
```suggestion
```

File: src/services/product-service.ts [hunk 2]
```diff
@@ -9,4 +10,15 @@ export class ProductService {
...
     return Api.post<IProduct>(apiUrl().products.create, newProduct);
   }
+  static deleteProduct(id: string) {
+    const product = products.find((p) => p.id === id);
+
+    if (!product) throw new Error("Product not found");
+    products.splice(
+      products.findIndex((p) => p.id === id),
+      1
+    );
```
Comment (amitmarko) [inline src/services/product-service.ts]: Please just make an api call...  
As we talked the delete should be on the MSW.
```suggestion
```

File: src/ui-kit/modal/DeleteModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+const deleteProduct = (id: string) => ProductService.deleteProduct(id);
+
+export const DeleteModal = ({ visible, onCancel, productId }: IProps) => {
+  const toast = useRef(null);
+
+  const { mutate } = useMutation(deleteProduct, {
+    onSuccess: () => ProductService.getAll(),
+  });
```
Comment (amitmarko) [inline src/ui-kit/modal/DeleteModal/index.tsx:43]: new line
```suggestion
  });
 
```

File: src/ui-kit/modal/DeleteModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  const onHide = () => {
```
Comment (amitmarko) [inline src/ui-kit/modal/DeleteModal/index.tsx]: useCallback

File: src/ui-kit/modal/DeleteModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    mutate(productId);
+    onCancel();
+    toast?.current?.show({
+      severity: "success",
+      summary: "Success Message",
+      detail: "Delete Success",
+      life: 3000,
+    });
```
Comment (amitmarko) [inline src/ui-kit/modal/DeleteModal/index.tsx]: If the API is failed what would happen? 
please move to `onSuccess`

File: src/ui-kit/modal/DeleteModal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        <Button
+          label="No"
+          icon="pi pi-times"
+          onClick={onCancel}
+          className="p-button-text"
+        />
+        <Button label="Yes" icon="pi pi-check" onClick={onHide} />
+      </div>
+    );
+  };
```
Comment (amitmarko) [inline src/ui-kit/modal/DeleteModal/index.tsx]:
```suggestion
  const renderFooter = () => (
      <div>
        <Button
          label="No"
          icon="pi pi-times"
          onClick={onCancel}
          className="p-button-text"
        />
        <Button label="Yes" icon="pi pi-check" onClick={onHide} />
      </div>
  );
```

File: src/ui-kit/modal/DeleteModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  return (
+    <div>
+      <Toast ref={toast} />
+      <Dialog
+        visible={visible}
+        onHide={onCancel}
+        footer={renderFooter()}
+        header="Delete Product"
+      >
+        <p>Are you sure that you want to delete product?</p>
```
Comment (amitmarko) [inline src/ui-kit/modal/DeleteModal/index.tsx:71]: Which Product? try to think UI/UX

File: src/ui-kit/modal/EditModal/index.tsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        <Button
+          label="No"
+          icon="pi pi-times"
+          onClick={onCancel}
+          className="p-button-text"
+        />
+        <Button label="Yes" icon="pi pi-check" onClick={onHide} autoFocus />
+      </div>
+    );
+  };
```
Comment (amitmarko) [inline src/ui-kit/modal/EditModal/index.tsx:23]: make common

File: src/ui-kit/modal/EditModal/index.tsx [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+  visible: boolean;
+  onCancel: () => void;
+}
+
+export const EditModal = ({ visible, onCancel }: IProps) => {
+  const [choose, setChoose] = useState(false);
+
+  const onHide = () => {
+    onCancel();
+  };
```
Comment (amitmarko) [inline src/ui-kit/modal/EditModal/index.tsx]: no need
```suggestion
  const onHide = () => {
    onCancel();
  };
```

File: src/ui-kit/dropdown/index.tsx [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+  const items = [
+    {
+      label: "Options",
+      items: [
+        {
+          label: "Edit",
+          icon: "pi pi-refresh",
+          command: () => {
+            // openModal("info", { productId });
```
Comment (amitmarko) [inline src/ui-kit/dropdown/index.tsx]:
```suggestion
```

File: src/ui-kit/modal/DeleteModal/index.tsx [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+  const { mutate } = useMutation(deleteProduct, {
+    onSuccess: () => ProductService.getAll(),
+  });
+  const onHide = () => {
+    mutate(productId);
+    onCancel();
+    toast?.current?.show({
+      severity: "success",
+      summary: "Success Message",
+      detail: "Delete Success",
```
Comment (amitmarko) [inline src/ui-kit/modal/DeleteModal/index.tsx]: Which product deleted? Please think UI/UX.

File: src/ui-kit/modal/EditModal/index.tsx [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+import { Dialog } from "primereact/dialog";
+import { Button } from "primereact/button";
+
+interface IProps {
+  visible: boolean;
+  onCancel: () => void;
+}
+
+export const EditModal = ({ visible, onCancel }: IProps) => {
+  const [choose, setChoose] = useState(false);
```
Comment (amitmarko) [inline src/ui-kit/modal/EditModal/index.tsx]: What is it???

File: src/services/product-service.ts [hunk 13]
```diff
@@ -1,3 +1,4 @@
+import { products } from "../mockserver/data/products";
```
Comment (amitmarko) [inline src/services/product-service.ts:1]:
```suggestion
```

File: src/Pages/Products/components/ProductsTable/index.tsx [hunk 14]
```diff
@@ -3,16 +3,24 @@ import { AgGridReact } from "@ag-grid-community/react";
 import styled from "styled-components";
 import { ProductService } from "../../../../services";
 import { productColumns } from "./productColumns";
+import { ModuleRegistry } from "@ag-grid-community/core";
+import { ClientSideRowModelModule } from "@ag-grid-community/client-side-row-model";
+import { useEffect, useLayoutEffect, useState } from "react";
+import { IProduct } from "models";
 
+ModuleRegistry.registerModules([ClientSideRowModelModule]);
```
Comment (amitmarko) [inline src/Pages/Products/components/ProductsTable/index.tsx:10]: What is it?

File: src/Pages/Products/components/ProductsTable/index.tsx (hunk 1 above)
Comment (maximFeldman1) [reply src/Pages/Products/components/ProductsTable/index.tsx]: Fixed

File: src/ui-kit/modal/DeleteModal/index.tsx (hunk 3 above)
Comment (maximFeldman1) [reply src/ui-kit/modal/DeleteModal/index.tsx:43]: Fixed

File: src/ui-kit/modal/DeleteModal/index.tsx (hunk 6 above)
Comment (maximFeldman1) [reply src/ui-kit/modal/DeleteModal/index.tsx]: Fixed

File: src/ui-kit/modal/EditModal/index.tsx (hunk 9 above)
Comment (maximFeldman1) [reply src/ui-kit/modal/EditModal/index.tsx]: Fixed

File: src/ui-kit/modal/EditModal/index.tsx (hunk 12 above)
Comment (maximFeldman1) [reply src/ui-kit/modal/EditModal/index.tsx]: Fixed

File: src/Pages/Products/components/ProductsTable/index.tsx (hunk 14 above)
Comment (maximFeldman1) [reply src/Pages/Products/components/ProductsTable/index.tsx:10]: without it,it doesnt show the table

File: src/services/product-service.ts (hunk 2 above)
Comment (maximFeldman1) [reply src/services/product-service.ts]: Fixed

File: src/ui-kit/modal/DeleteModal/index.tsx (hunk 4 above)
Comment (maximFeldman1) [reply src/ui-kit/modal/DeleteModal/index.tsx]: Fixed

File: src/ui-kit/dropdown/index.tsx (hunk 10 above)
Comment (maximFeldman1) [reply src/ui-kit/dropdown/index.tsx]: Fixed

File: src/ui-kit/modal/DeleteModal/index.tsx [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+}: IProps) => {
+  const toast = useRef(null);
+  const { mutate } = useMutation(deleteProduct, {
+    onSuccess: () => onDone(),
+  });
+
+  const onHide = useCallback(() => {
+    mutate(productId);
+    onCancel();
+    toast?.current?.show({
```
Comment (amitmarko) [inline src/ui-kit/modal/DeleteModal/index.tsx]: Please move toast to onSuccess function

File: src/ui-kit/modal/DeleteModal/index.tsx [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+
+export const DeleteModal = ({
+  visible,
+  onCancel,
+  productId,
+  onDone,
+}: IProps) => {
+  const toast = useRef(null);
+  const { mutate } = useMutation(deleteProduct, {
+    onSuccess: () => onDone(),
```
Comment (amitmarko) [inline src/ui-kit/modal/DeleteModal/index.tsx]: Please show toast message when there is an error

**Review by amitmarko — COMMENTED** [review]
comment

**Pull request merged**